### PR TITLE
Add "Search pat in mods" syntax equivalent to "Search pat inside mods"

### LINKED ISSUE #14930: Please accept `in` as a synonym of `inside` in `Search`
This would ease the transition of the old ssrsearch code.

### PR DESCRIPTION
**Pull request by SkySkimmer**
Close #14930

### CODE REVIEW

**Comment by SkySkimmer** [comment]
ping @coq/ssreflect-maintainers

**Review by gares — APPROVED** [review]

**Comment by gares** [comment]
I'm afraid you need a changelog entry since it is a user visible (and very appreciated) change

**Comment by SkySkimmer** [comment]
BTW in the issue there was mention of having `in:mod` (I guess 1 module, not a list), not sure how wanted that is.

**Comment by SkySkimmer** [comment]
@coq/nix-maintainers why is nix unhappy?

**Comment by gares** [comment]
Can you write `in:X in:Y`? That would be good enough IMO.

**Comment by gares** [comment]
I misread your comment, you implemented the list thing. I think it is OK (even if 99% of the times it is going to be a singleton).

**Comment by SkySkimmer** [comment]
Currently it's `in M` with no `:`

File: doc/changelog/07-vernac-commands-and-options/15139-search-in.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- **Added:**
+  :cmd:`Search` understands modifier ``in`` as an alias of ``inside``
```
Comment (jfehrle) [inline doc/changelog/07-vernac-commands-and-options/15139-search-in.rst:2]:
```suggestion
  :cmd:`Search` now accepts ``in`` as an alias of ``inside``
```

File: doc/changelog/07-vernac-commands-and-options/15139-search-in.rst [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  (`#15139 <https://github.com/coq/coq/pull/15139>`_,
+  fixes `#14930 <https://github.com/coq/coq/issues/14930>`_,
+  by Gaëtan Gilbert).
+  This is intended to ease transition for ssreflect Search users.
```
Comment (jfehrle) [inline doc/changelog/07-vernac-commands-and-options/15139-search-in.rst:6]: Maybe move this sentence before the ``` (`#15139 <https://github.com/coq/coq/pull/15139>`_, ```

File: doc/sphinx/proof-engine/vernacular-commands.rst [hunk 3]
```diff
@@ -324,7 +324,7 @@ described elsewhere
 
          SearchPattern (?X1 + _ = _ + ?X1).
 
-.. cmd:: SearchRewrite @one_pattern {? {| inside | outside } {+ @qualid } }
+.. cmd:: SearchRewrite @one_pattern {? {| {| inside | outside } {+ @qualid } | in {+ @qualid } } }
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/vernacular-commands.rst:327]: And also here:

image

File: doc/sphinx/proof-engine/vernacular-commands.rst [hunk 4]
```diff
@@ -299,7 +299,7 @@ described elsewhere
 
          Search is:Instance [ Reflexive | Symmetric ].
 
-.. cmd:: SearchPattern @one_pattern {? {| inside | outside } {+ @qualid } }
+.. cmd:: SearchPattern @one_pattern {? {| {| inside | outside } {+ @qualid } | in {+ @qualid } } }
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/vernacular-commands.rst:302]: Please update the following to include "in":

image

File: doc/sphinx/proof-engine/vernacular-commands.rst [hunk 5]
```diff
@@ -83,7 +83,7 @@ described elsewhere
    Displays the type of :n:`@term`. When called in proof mode, the
    term is checked in the local context of the selected goal.
 
-.. cmd:: Search {+ @search_query } {? {| inside | outside } {+ @qualid } }
+.. cmd:: Search {+ @search_query } {? {| {| inside | outside } {+ @qualid } | in {+ @qualid } } }
```
Comment (jfehrle) [inline doc/sphinx/proof-engine/vernacular-commands.rst:86]: This should appear like this for readability:

```suggestion
.. cmd:: Search {+ @search_query } {? {| inside | in | outside } {+ @qualid } }
```

To do so, change the `ne_in_or_out_modules` lines in `common_edit.mlg` as follows:

```
ne_in_or_out_modules: [
| REPLACE "inside" LIST1 global
| WITH [ "in" | "inside" | "outside" ] LIST1 global
| DELETE "in" LIST1 global
| DELETE "outside" LIST1 global
]
```

Also, please describe "in" in the documentation with "inside"

image

You could add another bullet point or use ``` :n:`{| inside | in }` ```

**Review by jfehrle — COMMENTED** [review]
Some tweaks, please.

File: doc/changelog/07-vernac-commands-and-options/15139-search-in.rst (hunk 1 above)
Comment (jfehrle) [reply doc/changelog/07-vernac-commands-and-options/15139-search-in.rst:2]: Or maybe say `Search*` without the `:cmd:` since 3 commands were changed.

Comment (Zimmi48) [reply doc/changelog/07-vernac-commands-and-options/15139-search-in.rst:2]: Or just list the three commands.

**Review by gares — APPROVED** [review]

**Comment by SkySkimmer** [comment]
Needs assignee

**Comment by gares** [comment]
@coqbot merge now

**Pull request merged**